clash-lib/src/Clash/Core/Pretty.hs

```diff
@@ -381,8 +382,8 @@ pprPrecCast prec e ty1 ty2 = do
     e' <> annotate (AnnSyntax Type)
                    (softline <> nest 2 (vsep [cast, ty1', coerce, ty2']))
 
-pprPrecLetrec :: Monad m => Rational -> [(Id, Term)] -> Term -> m ClashDoc
-pprPrecLetrec prec xes body = do
+pprPrecLetrec :: Monad m => Rational -> Bool -> [(Id, Term)] -> Term -> m ClashDoc
```

Comment:
I wonder whether we layout nonrec's differently from recs, because we can end up with a long chain of nonrecs when we create let-binders from pattern-binders in transformation like ANF, creating this large diagonal of let-expressions.

Reply:
This isn't so much of a problem right now, because a fair few transformations will still turn `let` back to `letrec` which get flattened (the end result of normalization typically still has a single `letrec` at the top). So unless you object I'd rather leave this until later

Reply:
This isn't so much of a problem right now, because a fair few transformations will still turn `let` back to `letrec` which get flattened (the end result of normalization typically still has a single `letrec` at the top). So unless you object I'd rather leave this until later

Reply:
Perhaps add a TODO then.